SU2_CFD/src/solvers/CMeshSolver.cpp

```diff
@@ -545,6 +545,63 @@ void CMeshSolver::DeformMesh(CGeometry **geometry, CNumerics **numerics, CConfig
 
 }
 
+void CMeshSolver::DeformMesh(CGeometry *geometry, CNumerics **numerics, CConfig *config){
+
+  if (multizone) nodes->Set_BGSSolution_k();
+
+  /*--- Capture a few MPI dependencies for AD. ---*/
+  geometry->InitiateComms(geometry, config, COORDINATES);
```

Comment:
So, what we do in such cases it to create a protected implementation function, e.g. DeformMesh_impl which in addition to a **geometry, etc., takes a boolean for MG, then the two DeformMesh overloads call the implementation function.

It does not have to be exactly like that, but the moral of the story is that if you ever see the need to copy-paste, cut-paste instead.
BTW I think this is a bit out of date with develop?